cmake/modules/AddSwift.cmake

```diff
@@ -1068,8 +1065,16 @@ function(_add_swift_library_single target name)
   set(link_flags ${SWIFTLIB_SINGLE_LINK_FLAGS})
   set(library_search_directories
       "${SWIFTLIB_DIR}/${SWIFTLIB_SINGLE_SUBDIR}"
-      "${SWIFT_NATIVE_SWIFT_TOOLS_PATH}/../lib/swift/${SWIFTLIB_SINGLE_SUBDIR}"
+      "${SWIFT_NATIVE_SWIFT_TOOLS_PATH}/../lib/swift/${SWIFTLIB_SINGLE_SUBDIR}")
+
+  if("${SWIFT_SDK_${SWIFTLIB_SINGLE_SDK}_OBJECT_FORMAT}" STREQUAL "ELF")
```

Comment:
Not sure if I should gate on elf or gate on mach. @compnerd windows?

Reply:
PE/COFF doesn't have fat binaries either.  I think that we should make this a Mach check.

Reply:
Line 1650 uses an `IS_DARWIN` check instead.

Reply:
Yeah, I can switch to that.